src/actions/actions.ts

```diff
@@ -5187,3 +5188,145 @@ class ActionOverrideCmdAltUp extends BaseCommand {
     return vimState;
   }
 }
+
+
+if (Configuration.getInstance().easymotion) {
+  @RegisterAction
+  class ActionEasyMotionCommand extends BaseCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "s", "<character>"];
+
+    public async exec(position: Position, vimState: VimState): Promise<VimState> {
+      const searchChar = this.keysPressed[3];
+      var searchState = new SearchState(SearchDirection.Forward, vimState.cursorPosition, searchChar);
+
+      var ranges = searchState.matchRanges;
+      if (ranges.length === 0) {
+        return vimState;
+      }
+
+      var filtered: EasyMotion.Match[] = [];
+
+      var totalLineLength = 0;
+      var prevLine = -1;
+      var cursorIndex = 0;
+      for (var i = 0; i < ranges.length; i++) {
+        var range = ranges[i];
+        var pos = Position.FromVSCodePosition(range.start);
+
+        if (pos.isEqual(position)) {
+          continue;
+        }
+
+        if (Math.abs(pos.line - position.line) > 100) {
+          continue;
+        }
+
+        if (prevLine === -1) {
+          prevLine = pos.line;
+        }
+        if (pos.line !== prevLine) {
+          const line = TextEditor.getLineAt(new Position(prevLine, 0)).text;
+          totalLineLength += line.length;
+        }
+
+        prevLine = pos.line;
+
+        if (position.line >= pos.line) {
+          cursorIndex = totalLineLength + position.character;
+        }
+
+        filtered.push(new EasyMotion.Match(pos, totalLineLength + pos.character));
+      }
+
+      filtered.sort((a: EasyMotion.Match, b: EasyMotion.Match): number => {
+        var diffA = cursorIndex - a.index;
+        var diffB = cursorIndex - b.index;
+
+        var absDiffA = Math.abs(diffA);
+        var absDiffB = Math.abs(diffB);
+
+        if (a.index > cursorIndex) {
```

Comment:
Can you comment all of your edge and corner cases? I don't know WHY a lot of these checks are here